packages/aws-cdk-lib/aws-appsync/lib/eventapi.ts

```diff
@@ -0,0 +1,584 @@
+    this.validateAuthorizationConfig(authProviders, defaultPublishAuthModeTypes);
+    this.validateAuthorizationConfig(authProviders, defaultSubscribeAuthModeTypes);
+
+    this.defaultPublishModeTypes = defaultPublishAuthModeTypes;
+    this.defaultSubscribeModeTypes = defaultSubscribeAuthModeTypes;
+
+    if (!Token.isUnresolved(props.ownerContact) && props.ownerContact !== undefined && props.ownerContact.length > 256) {
+      throw new Error('You must specify `ownerContact` as a string of 256 characters or less.');
+    }
+
+    this.eventConfig = {
+      authProviders: this.mapAuthorizationProviders(authProviders),
+      connectionAuthModes: this.mapAuthorizationConfig(connectionAuthModeType),
+      defaultPublishAuthModes: this.mapAuthorizationConfig(defaultPublishAuthModeTypes),
+      defaultSubscribeAuthModes: this.mapAuthorizationConfig(defaultSubscribeAuthModeTypes),
+      logConfig: this.setupLogConfig(props.logConfig),
+    };
+
+    this.api = new CfnApi(this, 'Resource', {
+      name: this.physicalName,
+      ownerContact: props.ownerContact,
+      eventConfig: this.eventConfig,
+    });
+
+    this.api.applyRemovalPolicy(RemovalPolicy.DESTROY);
+
+    this.apiId = this.api.attrApiId;
+    this.apiArn = this.api.attrApiArn;
+    this.dns = this.api.attrDns;
+
+    const apiKeyConfigs = authProviders.filter((mode) => mode.authorizationType === AuthorizationType.API_KEY);
+    for (const mode of apiKeyConfigs) {
+      this.apiKeys[mode.apiKeyConfig?.name ?? 'Default'] = createAPIKey(this, this.apiId, mode.apiKeyConfig);
+    }
+
+    if (authProviders.some((mode) => mode.authorizationType === AuthorizationType.LAMBDA)) {
+      const config = authProviders.find((mode: AuthorizationMode) => {
+        return mode.authorizationType === AuthorizationType.LAMBDA && mode.lambdaAuthorizerConfig;
+      })?.lambdaAuthorizerConfig;
+
+      config?.handler.addPermission(`${id}-appsync`, {
+        principal: new ServicePrincipal('appsync.amazonaws.com'),
+        action: 'lambda:InvokeFunction',
+        sourceArn: this.apiArn,
+      });
+    }
+
+    if (props.domainName) {
+      this.domainNameResource = new CfnDomainName(this, 'DomainName', {
+        domainName: props.domainName.domainName,
+        certificateArn: props.domainName.certificate.certificateArn,
+        description: `domain for ${props.apiName} Event API`,
+      });
+      const domainNameAssociation = new CfnDomainNameApiAssociation(this, 'DomainAssociation', {
+        domainName: props.domainName.domainName,
+        apiId: this.apiId,
+      });
+
+      domainNameAssociation.addDependency(this.domainNameResource);
+    }
+
+    const logGroupName = `/aws/appsync/apis/${this.apiId}`;
+
+    if (props.logConfig) {
+      const logRetention = new LogRetention(this, 'LogRetention', {
+        logGroupName: logGroupName,
+        retention: props.logConfig?.retention ?? RetentionDays.INFINITE,
```

Comment:
For my understanding, is `infinite` the right default?

Reply:
Correct - we generally include the logs are kept forever if the user doesn't explicitly provide a retention period. This way we're not deleting logs if the user didn't explicitly say to after a period of time. This is consistent with the AppSync GraphQL implementation [here](https://github.com/mazyu36/aws-cdk/blob/6a02ca19d74665a277cd9d48b56b0e2967a01424/packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts#L599).